src/timeseriesflattener/flattened_dataset.py

```diff
@@ -526,196 +615,174 @@ def _add_incident_outcome(
+            required_columns += [self.timestamp_col_name]
 
-        Args:
-            resolve_multiple (Callable): Takes a grouped df and collapses each group to one record (e.g. sum, count etc.).
-            df (DataFrame): Source dataframe with all prediction time x val combinations.
-            pred_time_uuid_colname (str): Name of uuid column in df.
+        for col in required_columns:
+            if col not in spec.values_df.columns:  # type: ignore
+                raise ValueError(f"Missing required column: {col}")
 
-        Returns:
-            DataFrame: DataFrame with one row pr. prediction time.
-        """
-        # Convert timestamp val to numeric that can be used for resolve_multiple functions
-        # Numeric value amounts to days passed since 1/1/1970
-        try:
-            df["timestamp_val"] = (
-                df["timestamp_val"] - dt.datetime(1970, 1, 1)
-            ).dt.total_seconds() / 86400
-        except TypeError:
-            log.info("All values are NaT, returning empty dataframe")
+    def add_spec(
+        self,
+        spec: Union[list[AnySpec], AnySpec],
+    ):
+        """Add a specification to the flattened dataset."""
+        if isinstance(spec, AnySpec):
+            specs_to_process: list[AnySpec] = [spec]
+        else:
+            specs_to_process = spec
 
-        # Sort by timestamp_pred in case resolve_multiple needs dates
-        df = df.sort_values(by="timestamp_val").groupby(pred_time_uuid_colname)
+        for spec_i in specs_to_process:
+            allowed_spec_types = (OutcomeSpec, PredictorSpec, StaticSpec)
 
-        if isinstance(resolve_multiple, str):
-            resolve_multiple = resolve_multiple_fns.get(resolve_multiple)
+            if not isinstance(spec_i, allowed_spec_types):
+                raise ValueError(
+                    f"Input is not allowed. Must be one of: {allowed_spec_types}",
+                )
 
-        if callable(resolve_multiple):
-            df = resolve_multiple(df).reset_index()
-        else:
-            raise ValueError("resolve_multiple must be or resolve to a Callable")
+            self._check_that_spec_df_has_required_columns(spec=spec_i)
 
-        return df
+            if isinstance(spec_i, OutcomeSpec):
+                self.unprocessed_specs.outcome_specs.append(spec_i)
+            elif isinstance(spec_i, PredictorSpec):
+                self.unprocessed_specs.predictor_specs.append(spec_i)
+            elif isinstance(spec_i, StaticSpec):
+                self.unprocessed_specs.static_specs.append(spec_i)
 
-    @staticmethod
-    def _drop_records_outside_interval_days(
-        df: DataFrame,
-        direction: str,
-        interval_days: float,
-        timestamp_pred_colname: str,
-        timestamp_value_colname: str,
-    ) -> DataFrame:
-        """Keep only rows where timestamp_value is within interval_days in
+    def add_age_and_birth_year(
+        self,
+        id2date_of_birth: DataFrame,
```

Comment:
id2date_of_birth is not a (1 to 1) mapping, therefore we shouldn't use the x2x name. I suggest date_of_birth_df